### ISSUE TITLE
Cannot connect to a CloudAMQP server

### ISSUE THREAD

**Issue by dfischer0** [arobson/rabbot#11]
I am trying to connect to a (free) CloudAMQP instance, and I am getting the following error: "Socket closed abruptly during opening handshake"
I am using the following URI: `amqp://uzhsuyri:<password_removed>@squirrel.rmq.cloudamqp.com/uzhsuyri`

Does rabbot support VHosts? CloudAMQP?

I tried pointing the demo/pubsub to that server, and it failed to connect as well. However, the sample at https://www.rabbitmq.com/tutorials/tutorial-one-javascript.html, which uses amqp.node, works perfectly...

**Comment by dfischer0** [comment]
And by the way, I also tested my code using the following URI: [amqp://guest:guest@localhost/](url) and it worked just fine.

**Comment by arobson** [comment]
@dfischer - are you passing the uri as the `uri` parameter or are you using separate config options? My guess is that either I have something incorrect in the way I'm parsing uri or you may be passing the vhost argument incorrectly.

**Comment by dfischer0** [comment]
@arobson - I am passing the uri as the `uri` parameter, and now that you suggested it, I tried passing separate parameters and it worked. In fact, you are right, the problem is in the way you parse the vhost part of the uri. When passing the vhost as a parameter, it is kept _as is_ in the vhost field, _without the encoded leading slash_. But when passing it inside the uri, it ends up in the vhost field _with_ the encoded leading slash.
So I'm assuming it a very simple fix. Do you think you could release it soon?
Thanks,
David

**Comment by dfischer0** [comment]
Ok, actually I guess the bug is that you are now expecting the vhost _without_ the leading slash in the vhost parameter, and you are adding it automatically.
Anyway, it's very simple (2 minutes really) to create a free account and queue on CloudAMQP if you want to test it.

**Comment by arobson** [comment]
@dfischer0 I haven't tested this against CloudAMQP, but I went back to the RabbitMQ docs and I see what I did wrong - I was using a standard url lib to parse and used the path as vhost. At one point it was convention/common practice to start vhosts with a leading `/` and I can't figure out why on earth they'd force you to escape each path delimiter for your vhost in a format that already supports it natively ...

Anyway. The defect is fixed now and published to NPM as 1.0.1.
